vendor/github.com/mongodb/mongo-tools-common/db/read_preferences.go

```diff
@@ -24,38 +25,62 @@ const (
 		"connection to mongos may produce inconsistent duplicates or miss some documents."
 )
 
-func Primary() *readpref.ReadPref          { return readpref.Primary() }
-func PrimaryPreferred() *readpref.ReadPref { return readpref.PrimaryPreferred() }
-func Nearest() *readpref.ReadPref          { return readpref.Nearest() }
+// NewReadPreference takes a string (command line read preference argument) and a ConnString (from the command line
+// URI argument) and returns a ReadPref. If both are provided, preference is given to the command line argument. If
+// both are empty, a default read preference of primary will be returned.
+func NewReadPreference(cmdReadPref string, cs *connstring.ConnString) (*readpref.ReadPref, error) {
+	// default to primary if a read preference isn't specified anywhere
+	if cmdReadPref == "" && (cs == nil || cs.ReadPreference == "") {
+		return readpref.Primary(), nil
+	}
+
+	var doc readPrefDoc
+	var err error
+	if cmdReadPref != "" {
+		doc, err = readPrefDocFromString(cmdReadPref)
+	} else {
+		doc = readPrefDocFromConnString(cs)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	mode, err := readpref.ModeFromString(doc.Mode)
+	if err != nil {
+		return nil, err
+	}
+
+	tagSet := tag.NewTagSetFromMap(doc.Tags)
+	return readpref.New(mode, readpref.WithTagSets(tagSet))
```

Comment:
We should also be parsing `maxStalenessMS`. See the [server selection spec](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#components-of-a-read-preference) for more info.